content/frontend/angular-apollo/2-queries-loading-links.md

```diff
@@ -0,0 +1,322 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Link} from '../types';
+
+@Component({
+  selector: 'hn-link-item',
+  templateUrl: './link-item.component.html',
+  styleUrls: ['./link-item.component.css']
+})
+export class LinkItemComponent implements OnInit {
+  @Input()
+  link: Link;
+
+  constructor() {
+  }
+
+  ngOnInit() {
+  }
+
+  voteForLink= async () => {
+    // ... you'll implement this in chapter 6
+  }
+}
+```
+
+```html(path=".../hackernews-angular-apollo/src/app/link-item/link-item.component.html")
+
+<div>{{link.description}} ({{link.url}})</div>
+
+```
+</Instruction>
+
+Note, we will be writing all our typings in a `./src/app/types.ts` file and simply importing these types into components as needed.
+
+<Instruction>
+
+This command will create a `link-item` folder containing several new files called `link-item.component.ts` , `link-item.component.spec.ts`, `link-item.component.html`, `link-item.component.css` in the `app` directory and add the following code in `link-item.component.ts`:
+
+```ts(path=".../hackernews-angular-apollo/src/app/type.ts")
+export class Link {
+  id: string;
+  description: string;
+  url: string;
+  createdAt: string;
+}
+
+```
+</Instruction>
+
+
+This is a simple Angular component that expects a `link` in its `Input` (`link-item.component.ts`) and renders the link's `description` and `url` (`link-item.component.html`). Easy as pie! 🍰
+
+Next, you'll implement the component that renders a list of link-items.
+
+<Instruction>
+
+Again, in the root directory, go ahead and run `ng generate component link-list` and this create `link-list.component.ts` , `link-list.component.spec.ts`, `link-list.component.html`, `link-list.component.css` in `src/app/link-list` folder and add the following code in `link-list.component.ts`:
+
+```ts(path=".../hackernews-angular-apollo/src/app/link-list/link-list.component.ts")
+import {Component, OnInit} from '@angular/core';
+import {Link} from '../types';
+
+@Component({
+  selector: 'hn-link-list',
+  templateUrl: './link-list.component.html',
+  styleUrls: ['./link-list.component.css']
+})
+export class LinkListComponent implements OnInit {
+  linksToRender: Link[] = [{
+    id: '1',
+    description: 'The Coolest GraphQL Backend 😎',
+    url: 'https://www.graph.cool'
+  }, {
+    id: '2',
+    description: 'The Best GraphQL Client',
+    url: 'http://dev.apollodata.com/'
+  }];
+
+  constructor() {
+  }
+
+  ngOnInit() {
+  }
+
+}
+```
+Then, add the following code in `link-list.component.html`:
+
+```html(path=".../hackernews-angular-apollo/src/app/link-list/link-list.component.html")
+
+<hn-link-item *ngFor="let link of linksToRender"
+          [link]="link">
+</hn-link-item>
+
+```
+
+</Instruction>
+
+
+Here, you're using mock data (`link-list.component.ts`) for now to make sure the component setup works. You'll soon replace this with some actual data loaded from the server - patience, young Padawan!
+
+<Instruction>
```

Comment:
same problem as before that this instruction block doesn't render properly, need to make two out of it